src/java.base/share/classes/java/util/zip/ZipInputStream.java

```diff
@@ -64,11 +64,17 @@
  *      }
  *    }
  * @apiNote
- * The LOC header contains metadata about the Zip file entry. {@code ZipInputStream}
- * does not read the Central directory (CEN) header for the entry and therefore
- * will not have access to its metadata such as the external file attributes.
- * {@linkplain ZipFile} may be used when the information stored within
- * the CEN header is required.
+ * The LOC header contains metadata about the Zip file entry.
+ * {@code ZipInputStream} does not read the Central directory (CEN) header for
+ * the entry and therefore will not have access to its metadata such as the
+ * external file attributes. Additionally, {@code ZipInputStream} might read
+ * entries that are not in the Central directory or contain information that is
+ * different than in the Central directory (CEN) header for the same entry.
+ * This class might also fail to properly parse ZIP archives that have
+ * prepended data.
+ *
+ * Whenever possible, {@linkplain ZipFile} should be used for parsing ZIP
+ * archives since it correctly reads data from the central directory.
```

Comment:
I think it's okay to extend the existing API note to say that the stream may contain ZIP file entries that are not are not named in the central directory.  I think I would replace the sentence "Additionally ..." with "Consequently, a ZipInputStream that reads from a ZIP file may read ZIP file entries that are not in the ZIP file's central directory". 

The sentence "This class might also fail to properly parse ZIP archives that have prepended data" will confuse readers as it hints of unreliability in scenarios and begs too many questions on where the data is prepended. So I think drop this unless you can come up with something better.

Replacing the sentence "ZipFile may be used ..." is okay but I don't think the proposed text works. The API note starts with text to say that a ZipOutputStream doesn't read the CEN so saying "correctly reads from the central directory" is very confusing. ZipInputStream is a JDK 1.1 era API, it is what it is and would be disruptive to deprecate.